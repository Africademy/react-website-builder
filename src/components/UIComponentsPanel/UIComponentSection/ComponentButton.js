import React from 'react'
const { Button } = require("react-bootstrap");

class ComponentButton extends React.Component{
    render(){
        return <Button className="m-2" variant="outline-success">{this.props.name}</Button>
    }
}

export default ComponentButton;