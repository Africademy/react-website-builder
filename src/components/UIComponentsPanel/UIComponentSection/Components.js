import React from "react";
class Components extends React.Component {
  render() {
    return <div></div>;
  }
}

export default Components;
