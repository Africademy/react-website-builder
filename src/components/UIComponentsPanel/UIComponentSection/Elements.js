import React from "react";
const { Button } = require("react-bootstrap");

class Elements extends React.Component {
  constructor(props) {
    super(props);

    this.dragstart_handler = this.dragstart_handler.bind(this);
  }

  dragstart_handler(e) {
    e.dataTransfer.setData("elementId", e.target.id);
    var attr = JSON.stringify(this.props.attr);
    e.dataTransfer.setData("attributes", attr);
  }

  render() {
    return (
      <Button
        className="m-2"
        draggable="true"
        onDragStart={this.dragstart_handler}
        variant="outline-success"
      >
        {this.props.name}
      </Button>
    );
  }
}

export default Elements;
