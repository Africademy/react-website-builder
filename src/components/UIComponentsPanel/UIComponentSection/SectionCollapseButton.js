import React from 'react'
import { Button } from 'react-bootstrap'

class SectionCollapseButton extends React.Component{
    render(){
        return (
            <Button block 
                    className = 'section-btn'
                    aria-controls={this.props.name} 
                    aria-expanded={true}
                    onClick = {this.props.onClick}>
                    {this.props.name}
            </Button>
        )
    }
}

export default SectionCollapseButton;