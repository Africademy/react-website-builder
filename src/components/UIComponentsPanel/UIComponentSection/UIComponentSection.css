.section-btn{
    cursor:default !important;

    color: #000 !important; 
    background-color: white !important; 
    border-color: white !important;
    border-bottom-color: black !important;
    box-shadow: none!important;
    outline: 0;
}

.component-btn{
    cursor:default !important;

    color: #000 !important; 
    background-color: white !important; 
    border-color: white !important;
    border-bottom-color: black !important;
    box-shadow: none!important;
    outline: 0;
}