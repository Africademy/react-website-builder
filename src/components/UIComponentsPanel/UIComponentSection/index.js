import React from 'react'
import ComponentButton from './ComponentButton'
import SectionCollapseButton from './SectionCollapseButton'
import * as  ReactBootstrap from 'react-bootstrap'
import './UIComponentSection.css'

class UIComponentSection extends React.Component {
    constructor(props){
        super(props)
        this.state = {isExpanded : true}
        this.handleExpansion = this.handleExpansion.bind(this)
    }

    handleExpansion(e){
        this.setState((state) => ({isExpanded : !state.isExpanded }));
    }

    render() {
        const rows = []
        const len = this.props.values.length;
        for (var i = 0; i < len/3; i++) {
            const row = []
            for(var j=i*3,k=0;j<len && k<3;j++,k++)
                row.push(<ComponentButton name={this.props.values[j]}></ComponentButton>);
            rows.push(<ReactBootstrap.Row className = "justify-content-center">{row}</ReactBootstrap.Row>);
        }
        
        return (
            <div className = "section">
                <SectionCollapseButton name={this.props.name} 
                                        onClick = {this.handleExpansion}
                />
                <ReactBootstrap.Collapse in={this.state.isExpanded}>
                    <ReactBootstrap.Container id={this.props.name}>{rows}</ReactBootstrap.Container>
                </ReactBootstrap.Collapse>
            </div>
        )
    }
}

export default UIComponentSection;