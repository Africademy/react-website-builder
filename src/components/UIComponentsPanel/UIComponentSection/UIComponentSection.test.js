import React from 'react';
import UIComponentSection from '.';
import Adapter from 'enzyme-adapter-react-16';
import { shallow,configure  } from 'enzyme';

const vals = {
    section1: {
        name: "Comps",
        values: [
            "A",
            "B",
            "C",
            "A",
            "B",
            "C"
        ]
    },
    section2: {
        name: "Others",
        values: [
            "A",
            "B"
        ]
    }
}
configure({adapter: new Adapter()});

test("show/hide components when section is clicked", () => {
    for (const key in vals) {
        if (vals.hasOwnProperty(key)) {
            const tree = shallow(
                <UIComponentSection key={vals[key].name} name={vals[key].name} values={vals[key].values}></UIComponentSection>
            );
            // let tree = comp.toJSON();
            // expect(tree).toMatchSnapshot();
            
            tree.find('SectionCollapseButton').prop('onClick')()
            expect(tree).toHaveBeenCalled();
        }
    }
});