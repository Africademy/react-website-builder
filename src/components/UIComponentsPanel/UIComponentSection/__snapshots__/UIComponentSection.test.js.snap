// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show/hide components when section is clicked 1`] = `
<div
  className="section"
>
  <button
    aria-controls="Comps"
    aria-expanded={true}
    className="section-btn btn btn-primary btn-block"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Comps
  </button>
  <div
    aria-expanded={null}
    className="collapse show container"
    id="Comps"
  >
    <div
      className="justify-content-center row"
    >
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        A
      </button>
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        B
      </button>
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        C
      </button>
    </div>
    <div
      className="justify-content-center row"
    >
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        A
      </button>
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        B
      </button>
      <button
        className="m-2 btn btn-outline-success"
        disabled={false}
        type="button"
      >
        C
      </button>
    </div>
  </div>
</div>
`;

exports[`show/hide components when section is clicked 2`] = `ShallowWrapper {}`;
